src/ng/q.js

```diff
@@ -677,10 +677,11 @@ function qFactory(nextTick, exceptionHandler, errorOnUnhandledRejections) {
 }
 
 function isStateExceptionHandled(state) {
+  if (!state) return true;
```

Comment:
Hm...I was going through the code looking for other places that might need fixing and I realized I am not why this should be necessary (i.e. where this function would be called from when using a 3rd-party promise library) 😕

Could you share a minimal demo (e.g. plnkr, repo, or live app (with non-minified sources :grin:)) that showcaes the current failures?

Reply:
It seems like this change is indeed not necessary ([relevant discussion][1]).

[1]: https://github.com/angular/angular.js/issues/16164#issuecomment-392356316

Reply:
> Hm...I was going through the code looking for other places that might need fixing and I realized I am not why this should be necessary (i.e. where this function would be called from when using a 3rd-party promise library) 😕
> 
> Could you share a minimal demo (e.g. plnkr, repo, or live app (with non-minified sources 😁)) that showcaes the current failures?

We are also using using Bluebird.js + angular-bluebird-promises. Setting ngModelOptions raises the TypeError: Cannot set property 'pur' of undefined. Please find a demo here http://plnkr.co/edit/9oDzLt?p=preview

Reply:
Thx for the reproduction, @davidwilder. Based on the stacktrace, the error happens in `markQStateExceptionHandled()` method and not the `isStateExceptionHandled()`. As mentioned in https://github.com/angular/angular.js/issues/16164#issuecomment-392455464, I don't think `isStateExceptionHandled()` needs to be changed (but it won't hurt either).